From 053bac34b9f0f947c58fcdf80ac03d5a6b05aa12 Mon Sep 17 00:00:00 2001
From: "Yann E. MORIN" <yann.morin.1998@free.fr>
Date: Mon, 7 May 2018 23:15:58 +0200
Subject: [PATCH] mbox: do not use backtrace if using uClibc without backtrace
 support

Since uClibc can be configured without support for backtrace, disable
the backtrace if we are building with a uClibc that was built without
backtrace.

This is a bit hacky, and would greatly benefit from a test in ./configure
instead, but does nicely as a quick fix for now.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
[Bernd: rebased for 0.103.0]
---
 libclamav/mbox.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libclamav/mbox.c b/libclamav/mbox.c
index 71f540c0a..4851297a9 100644
--- a/libclamav/mbox.c
+++ b/libclamav/mbox.c
@@ -91,7 +91,7 @@
 #include <features.h>
 #endif
 
-#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 1
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ >= 1 && !defined(__UCLIBC__) || defined(__UCLIBC_HAS_BACKTRACE__)
 #define HAVE_BACKTRACE
 #endif
 #endif
-- 
2.14.3

